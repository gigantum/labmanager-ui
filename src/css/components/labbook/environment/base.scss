.Base *{
  font-weight: 100;

}

.Base__header-container{
  @include flex(left, row)
}

.Base__header{
  color: $romanSilver;
}

.Base__card{
  color: $romanSilver;
  @extend .card;
  @include flex(space-between, row);
  img{
    min-width: 70px;
    margin: 0 10px 0 0;
  }

  padding: 30px;
}


.Base__image-container{
  width: 250px;
  margin: 0 20px 0 0;
  @include flex(space-between, row);

  p{
    font-size: 14px;
    font-weight: 400;
  }

  .Base__name{
    font-weight: 400;
  }

  .Base__title{
    flex: 1;
    width: 160px;
  }
}

.Base__card-text{
  flex: 1;

  p{
    margin-top: 0px;
  }
}

.Base__categories{
  @include flex(flex-start, row);

  h6{
    font-size: 16px;
    font-weight: 400;
  }
}

.Base__categories-languages{
  width: 50%;
}

.Base__categories-tools{
  width: 50%;
}
