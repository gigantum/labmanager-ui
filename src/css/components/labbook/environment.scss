.Environment{
  overflow: auto;
  height: calc(100% - 13px);
  width: 100%;
}

.Environment__header-container{
  @include flex(left, row)
}

.Environment__header{
  font-weight: light;
  color: $font-grey;
  font-size: 22px;
}

.Environment__edit-button,
.Environment__edit-button:hover,
.Environment__edit-button:focus{
  font-weight: light;
  color: $font-grey;
  font-size: 14px;
  height: 30px;
  width: 30px;
  padding: 5px;
  border: 0px;
  background: url('./../images/icons/edit.svg') no-repeat center center;
  background-size: 22px;
}

.Environment__card{
  background-color: $white;
  padding: 10px;
  color: $charcoal;
  width: 330px;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 1px 5px 0 rgba(0, 0, 0, 0.12), 0 3px 1px -2px rgba(0, 0, 0, 0.2);
}


.Environment__image-container{
  align-items: center;
  align-content: center;
  align-self: center;
}

.Environment__card-text{
  //background-color: $grey-light;
  padding: 10px;
  width: 200px;
}

.Environment__edit-container{
  align-self: center;
  margin-left: 20px;
}

.Environment__modal{
  position: fixed;
  left: 50%;
  top: 50%;
  z-index: 100;
  width: 800px;
  height: 600px;

  margin: -300px 0 0 -400px;

  border: 1px solid $grey;
  background-color: $grey-light;
  box-shadow: 10px 10px 5px #888888;

  .SelectBaseImage__progress-button{
    display: none;
  }

}

.Environment__modal-close{
  position: absolute;
  top: 5px;
  right: 5px;
  cursor: pointer;
}

.Environment__package-dependencies{
  margin: 10px 20px 10px 0;

  .Environment__card-text{
      background-color: $white;
      width: 150px;
      box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 1px 5px 0 rgba(0, 0, 0, 0.12), 0 3px 1px -2px rgba(0, 0, 0, 0.2);
      p{
        margin: 0 5px 0 0;
      }

  }
}

.Environment__development-environment-item,
.Environment__dependencies{
  margin: 0 20px 0 0;
}
