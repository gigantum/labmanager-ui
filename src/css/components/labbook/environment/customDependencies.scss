.CustomDependencies{
  position: relative;
}

.CustomDependencies__card{
  @extend .card;
  padding: 20px;
}

.CustomDependencies__button--flat,
.CustomDependencies__button--flat:focus{
  padding: 5px 0;
  background-color: $grey-light;
  border: none;
  color: $azure;
  width: 100%;
}

.CustomDependencies__button--flat:hover{
  background-color: $grey-light;
  border: none;
  color: $turquise;
  width: 100%;
}

.CustomDependencies__add-dependencies{
  background-color: $grey-light;
  border-radius: 8px;
  padding: 5px 20px;
  height: auto;
  margin: 20px 0;
}

input.CustomDependenciesDropdown__input[type="text"]{
  border: 1px solid $romanSilver;
  border-radius: 8px;
  padding: 0 15px;
  width: calc(100% - 30px);
  background: url(../images/icons/down-arrow-silver.svg) no-repeat 95% 50%;
  background-size: 20px;
  cursor: pointer;
}
input.CustomDependenciesDropdown__input[type="text"]:focus{
  cursor: text;
}

.CustomDependenciesDropdown__menu{

  @extend .box-shadow;
  padding: 20px;
  top: 142px;
  position: absolute;
  overflow: hidden;
  height: auto;
  max-height: 210px;
  transition: height 0.5s;

  z-index: 200;

  background: $white;
  width: calc(100% - 118px);
}

.CustomDependenciesDropdown__menu-row{
  @include flex(space-between, row);
  width: 100%;
  height: 40px;
  border-bottom: 1px solid $romanSilver;
  align-items: center;
}


.CustomDependenciesDropdown__menu--no-height{
  position: absolute;
  height: 0px;
  padding: 0px;
}

.CustomDependenciesDropdown__button--round,
.CustomDependencies__button--round{
  width: 30px;
  height: 30px;
  border-radius: 15px;
  padding: 0;
}
.CustomDependenciesDropdown__button--add,
.CustomDependencies__button--add{
  background: $azure url('../images/icons/add.svg') no-repeat 50% 50%;
  background-size: 16px;
}
.CustomDependenciesDropdown__button--add:hover,
.CustomDependencies__button--add:hover{
  background: $turquise url('../images/icons/add.svg') no-repeat 50% 50%;
  background-size: 16px;
}

.CustomDependencies__button--remove{
  background: $azure url('../images/icons/close-white.svg') no-repeat 50% 50%;
  background-size: 14px;
}
.CustomDependencies__button--remove:hover{
  background: $turquise url('../images/icons/close-white.svg') no-repeat 50% 50%;
  background-size: 14px;
}

.CustomDependencies__table-container{
  position: relative;
  padding: 20px 20px 60px;
  border: 1px solid $romanSilver;

  .CustomDependencies__button{
    position: absolute;
    bottom: -20px;
    left: 50%;
    width: 240px;
    height: 40px;
    margin: 0 0 0 -120px;
  }
}

.CustomDependencies__item{
  height: 40px;
  @include flex(space-between, row);

  border-bottom: 1px solid $romanSilver;
}

.CustomDependencies__view-container{
  padding: 20px 20px 40px;
  height: auto;
  overflow: hidden;
}

.CustomDependencies__view-container--no-height{
  height: 0px;
  max-height: 0px;
  padding: 0px;
}
