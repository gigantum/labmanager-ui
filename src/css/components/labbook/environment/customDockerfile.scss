.CustomDockerfile {
  position: relative;
  hr {
    opacity: 0.5;
    max-width: 85%;
    margin: 0;
  }
}

.CustomDockerfile__header {
  color: $romanSilver;
}

.CustomDockerfile__sub-header {
  margin-top: 0;
  font-size: 14px;
  color: $romanSilver;
  p{
    margin: 5px 0;
  }
  p:first-child {
    margin: 0 0 10px 0;
  }
  code {
    font-family: monospace;
  }
}

.CustomDockerfile--code-snippet code{
  padding: 2px 10px;
  margin: -4px 0 4px 11px;
 }

.CustomDockerfile__content {
  @extend .card;
  padding: 10px !important;
  color: $jet;
  white-space: pre-wrap;
  padding: 0;
  word-wrap: break-word;
  justify-content: space-between;
  flex-wrap: nowrap;
  flex-direction: column;

  p {
    margin: 0
  }

  div:first-child {

    /*
      have to use extend as ReactMarkDown does not inherit classNames
    */

    margin-bottom: 0;
    p {
      font-family: monospace !important;
      font-size: 1em !important;
    }
  }
  button {
    margin-top: 2px;
  }
  .column-1-span-2 {
    align-self: flex-end;

    @include flex(space-between, row);
  }
  .empty {
    color: $romanSilver;
    opacity: 0.5;
    font-style: oblique;
  }
  .column-1-span-11{
    align-self: center;
  }
}

.CustomDockerfile__content-input {
  height: 200px;
  font-family: monospace;
  font-size: 1em;
}

.CustomDockerfile__content-input:focus {
  border: 2px solid #26a69a;
  box-shadow: 0 1px 0 0 #26a69a;
  outline: none;
}

.CustomDockerfile__content-edit-button,
.CustomDockerfile__content-edit-button:hover,
.CustomDockerfile__content-edit-button:focus,
.CustomDockerfile__content-edit-button:disabled {
  height: 22px;
  width: 22px;
  border: 0px;
  background: url('./../images/icons/edit.svg') no-repeat center center;
  background-size: 22px;
}
.CustomDockerfile__content-edit-button:disabled {
  opacity: 0.2;
}

.CustomDockerfile__content-cancel-button,
.CustomDockerfile__content-cancel-button:hover,
.CustomDockerfile__content-cancel-button:focus {
  padding: 0;
}

.CustomDockerfile__button-container{
  @include flex(flex-end, row);
}