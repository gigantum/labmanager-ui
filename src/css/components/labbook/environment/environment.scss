.Environment {
  width: 100%;
}

.Environment__header-container {
  @include flex(left, row)
}

.Environment__header {
  font-weight: light;
  color: $font-grey;
  font-size: 22px;
}

.Environment__edit-button,
.Environment__edit-button:hover,
.Environment__edit-button:focus {
  font-weight: light;
  color: $font-grey;
  font-size: 14px;
  height: 30px;
  width: 30px;
  padding: 5px;
  border: 0px;
  background: url('./../images/icons/edit.svg') no-repeat center center;
  background-size: 22px;
}

.Environment__card {
  background-color: $white;
  padding: 10px;
  color: $jet;
  width: 100%;
  @extend .box-shadow;
}


.Environment__image-container {
  align-items: center;
  align-content: center;
  align-self: center;
}

.Environment__card-text {
  //background-color: $grey-light;
  padding: 10px;
  width: 200px;
}

.Environment__edit-container {
  align-self: center;
  margin-left: 20px;
}

.Environment__modal {
  position: fixed;
  @include modal(500px, 800px);

  .SelectDevelopmentEnvironment__title,
  .AddEnvironmentPackage__title {
    font-size: 1.5em;
    text-align: center;
    border-bottom: 1px solid $azure;
    padding: 5px;
    color: $azure;
  }

  .SelectBaseImage__progress-button {
    display: none;
  }

}

.Environment__modal-close {
  cursor: pointer;
  @include close();
}

.Environment__package-dependencies {
  margin: 10px 20px 10px 0;

  .Environment__card-text{
      background-color: $white;
      width: 150px;
      @extend .box-shadow;

      p{
        margin: 0 5px 0 0;
      }

  }
}

.Environment__development-environment-item,
.Environment__dependencies {
  margin: 0 20px 0 0;
}
