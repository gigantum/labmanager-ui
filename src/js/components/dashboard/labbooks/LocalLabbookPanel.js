//vendor
import React, { Component } from 'react'
//mutations
import ExportLabbookMutation from 'Mutations/ExportLabbookMutation'
//utilities
import JobStatus from 'JS/utils/JobStatus'
//store
import store from 'JS/redux/store'

/**
*  labbook panel is to only render the edge passed to it
*/

export default class LocalLabbookPanel extends Component {

  constructor(props) {
    super(props);

    this.state = {
      'exportPath': '',
    }

    this._exportLabbook = this._exportLabbook.bind(this)
  }

  _getContainerStatusText(containerStatus, imageStatus){

    let status = (containerStatus === 'RUNNING') ? 'Open' : containerStatus;
    status = (containerStatus === 'NOT_RUNNING') ? 'Closed' : status;
    status = (imageStatus === "BUILD_IN_PROGRESS") ? 'Building' : status;

    return status;
  }
  /**
  * @param {event, object} evt,edge
  *  runs export mutation if export has not been downloaded
  *
  */
  _exportLabbook = (evt, edge) => {

    if(this.state.exportPath.length === 0){

      let exportClassList = document.getElementById(evt.target.id).classList;

      exportClassList.add('LocalLabbooks__export--downloading')
      let username = localStorage.getItem('username')
      let labbookName = edge.node.name
      store.dispatch({
        type: 'INFO_MESSAGE',
        payload: {
          message: `Exporting ${labbookName} LabBook`,
        }
      })

      ExportLabbookMutation(username, edge.node.name, (response, error)=>{

        if(response.exportLabbook){
          JobStatus.getJobStatus(response.exportLabbook.jobKey).then((data)=>{


              if(data.jobStatus.result){
                store.dispatch({
                  type: 'INFO_MESSAGE',
                  payload: {
                    message: `Export file ${data.jobStatus.result} is available in the export directory of your Gigantum working directory.`,
                  }
                })
              }

              exportClassList.remove('LocalLabbooks__export--downloading')
          }).catch((error)=>{

              if(error){
                store.dispatch({
                  type: 'ERROR_MESSAGE',
                  payload: {
                    message: `${labbookName} failed to export `,
                    messageList: error
                  }
                })
              }
              exportClassList.remove('LocalLabbooks__export--downloading')
          })
      }else{

        store.dispatch({
          type: 'ERROR_MESSAGE',
          payload: {
            message: 'Export Failed: ' + error[0].message,
            messageList: error
          }
        })

        exportClassList.remove('LocalLabbooks__export--downloading')
      }
    })
  }

  }


  render(){
    let edge = this.props.edge;
    let status = this._getContainerStatusText(edge.node.environment.containerStatus, edge.node.environment.imageStatus)

    return (
      <div
        key={edge.node.name}
        className='LocalLabbooks__panel flex flex--column justify--space-between'>

        <div className="LocalLabbooks__icon-row">

          <div className="LocalLabbooks__containerStatus">
            <div className={'LocalLabbooks__containerStatus--state ' + status}>
              {status}
            </div>
          </div>
        </div>

        <div className="LocalLabbooks__text-row">
          <div className="LocalLabbooks__title-row">
            <h4
              className="LocalLabbooks__panel-title"
              onClick={() => this.props.goToLabbook(edge.node.name, edge.node.owner.username)}>
              {edge.node.name}
            </h4>
            <div className="LocalLabbooks__edit-button" onClick={() => this.props.renameLabbookModal(edge.node.name)}>
            </div>
          </div>
          <p className="LocalLabbooks__owner">{'Created by ' + edge.node.owner.username}</p>
          <p
            onClick={() => this.props.goToLabbook(edge.node.name, edge.node.owner.username)} className="LocalLabbooks__description">
            {edge.node.description}
          </p>
        </div>

        <div className="LocalLabbooks__info-row flex flex--row justify--space-between">
          <div className="LocalLabbooks__owner flex flex--row">
              {/* <div> {owner.username}</div> */}

          </div>
          <div className="LocalLabbooks__status">
            <div
              id={'Export__localLabbooks' + edge.node.name}
              ref={'Export__localLabbooks' + edge.node.name}
              onMouseDown={(evt) => this._exportLabbook(evt, edge)} className="LocalLabbooks__export">
              Export
            </div>

          </div>

        </div>


    </div>)
  }
}
