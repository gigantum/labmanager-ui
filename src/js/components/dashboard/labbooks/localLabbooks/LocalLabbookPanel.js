//vendor
import React, { Component } from 'react'
import Highlighter from 'react-highlight-words'
import {Link} from 'react-router-dom'
//muations
import StartContainerMutation from 'Mutations/StartContainerMutation'
import StopContainerMutation from 'Mutations/StopContainerMutation'
//store
import store from 'JS/redux/store'

/**
*  labbook panel is to only render the edge passed to it
*/

export default class LocalLabbookPanel extends Component {

  constructor(props) {
    super(props);

    this.state = {
      'exportPath': '',
      'status': 'loading',
      'textStatus': '',
      'labbookName': props.edge.node.name,
      'owner': props.edge.node.owner
    }
    this._getContainerStatusText = this._getContainerStatusText.bind(this)
    this._stopStartContainer = this._stopStartContainer.bind(this)
  }
  /***
  * @param {Object} nextProps
  * processes container lookup and assigns container status to labbook card
  */
  UNSAFE_componentWillReceiveProps(nextProps) {
    let {environment} = nextProps.node
    if(environment){
    let status = this._getContainerStatusText(environment.containerStatus, environment.imageStatus)

    this.setState({status: status, textStatus: status})
    }
  }
  /***
  * @param {}
  * if environment exists when components will mount it will populate container status
  */
  UNSAFE_componentWillMount(){
    let {environment} = this.props.node
    if(environment){
      let status = this._getContainerStatusText(environment.containerStatus, environment.imageStatus)

      this.setState({status: status, textStatus: status})
    }
  }
  /***
  * @param {string, string} containerStatus, imageStatus
  * returns corrent container status by checking both container and imagestatus
  */
  _getContainerStatusText(containerStatus, imageStatus){

    let status = 'Running';
    status = (containerStatus === 'NOT_RUNNING') ? 'Stopped' : status;
    status = (imageStatus === "BUILD_IN_PROGRESS") ? 'Building' : status;
    status = (imageStatus === "BUILD_FAILED") ? 'Rebuild' : status;
    status = (imageStatus === "DOES_NOT_EXIST") ? 'Rebuild' : status;

    return status;
  }
  /***
  * @param {string} status
  * fires when a componet mounts
  * adds a scoll listener to trigger pagination
  */
  _stopStartContainer(evt, status){
    evt.preventDefault()
    evt.stopPropagation();
    evt.nativeEvent.stopImmediatePropagation();
    if(status === "Stopped"){
      this._startContainerMutation()
    }else if(status === "Running"){
      this._stopContainerMutation()
    } else if(status === "loading"){
      store.dispatch({
        type: 'INFO_MESSAGE',
        payload:{
          message: `Container status is still loading. The status will update when it is available.`
        }
      })
    }else if(status === "Building"){
      store.dispatch({
        type: 'INFO_MESSAGE',
        payload:{
          message: `Container is still building and the process may not be interrupted.`
        }
      })
    } else {
      store.dispatch({
        type: 'INFO_MESSAGE',
        payload:{
          message: `Container must be rebuilt. Open Project first and then try to run again.`
        }
      })
    }
  }
  /***
  * @param {string} status
  * starts labbook conatainer
  */
  _startContainerMutation(){
    let self = this;

    const {owner, labbookName} = this.state

    store.dispatch({
      type: 'INFO_MESSAGE',
      payload:{
        message: `Starting ${labbookName} container`
      }
    })

    this.setState({'status': 'Starting', textStatus: 'Starting'})

    StartContainerMutation(
      labbookName,
      owner,
      'clientMutationId',
      (response, error) =>{

        if(error){

          store.dispatch({
            type: 'ERROR_MESSAGE',
            payload:{
              message: `There was a problem starting ${this.state.labbookName}, go to Project and try again`,
              messageBody: error
            }
          })

          self.setState({textStatus: "Stopped", status: "Stopped"})

        }else{

          self.props.history.replace(`../../projects/${owner}/${labbookName}`)
        }
      }
    )
  }
  /***
  * @param {string} status
  * stops labbbok conatainer
  */
  _stopContainerMutation(){

    const {owner, labbookName} = this.state

    let self = this

    store.dispatch({
      type: 'INFO_MESSAGE',
      payload:{
        message: `Stopping ${labbookName} container`
      }
    })

    this.setState({'status': 'Stopping', textStatus: 'Stopping'})

    StopContainerMutation(
      labbookName,
      owner,
      'clientMutationId',
      (response, error) =>{

        if(error){

          console.log(error)

          store.dispatch({
            type: 'ERROR_MESSAGE',
            payload:{
              message: `There was a problem stopping ${this.state.labbookName} container`,
              messageBody: error
            }
          })

          self.setState({textStatus: "Running", status: "Running"})

        }else{

          this.setState({'status': 'Stopped', textStatus: 'Stopped'})
        }

      }
    )
  }
  /***
  * @param {object,string} evt,status
  * stops labbbok conatainer
  ***/
  _updateTextStatusOver(evt, status){

    let newStatus = status;

    if(status !== 'loading'){

      newStatus = (status === "Running") ? 'Stop' : newStatus;
      newStatus = (status === "Stopped") ? 'Run' : newStatus;
      this.setState({textStatus: newStatus})
    }
  }
  /***
  * @param {objectstring} evt,status
  * stops labbbok conatainer
  ***/
  _updateTextStatusOut(evt, status){

    if(status !== 'loading'){

      this.setState({textStatus: status})
    }
  }

  render(){
    let edge = this.props.edge;
    let status = this.state.status
    let textStatus = this.state.textStatus


    return (
      <Link
        to={`/projects/${edge.node.owner}/${edge.node.name}`}
        onClick={() => this.props.goToLabbook(edge.node.name, edge.node.owner)}
        key={'local' + edge.node.name}
        className='LocalLabbooks__panel column-4-span-3 flex flex--column justify--space-between'>

        <div className="LocalLabbooks__icon-row">

          <div className="LocalLabbooks__containerStatus">

            <button
              onClick={(evt)=> this._stopStartContainer(evt, status)}
              onMouseOver={(evt)=> this._updateTextStatusOver(evt, status)}
              onMouseOut={(evt)=> this._updateTextStatusOut(evt, status)}
              className={`LocalLabbooks__containerStatus--state ${status}`}>
              {textStatus}
            </button>

          </div>

        </div>

        <div className="LocalLabbooks__text-row">

          <div className="LocalLabbooks__title-row">

            <h6
              className="LocalLabbooks__panel-title"
              onClick={() => this.props.goToLabbook(edge.node.name, edge.node.owner)}>

              <Highlighter
                highlightClassName='LocalLabbooks__highlighted'
                searchWords={[store.getState().labbookListing.filterText]}
                autoEscape={false}
                caseSensitive={false}
                textToHighlight={edge.node.name}
              />

            </h6>

          </div>

          <p className="LocalLabbooks__owner">{'Created by ' + edge.node.owner}</p>

          <p
            className="LocalLabbooks__description">

            <Highlighter
              highlightClassName='LocalLabbooks__highlighted'
              searchWords={[store.getState().labbookListing.filterText]}
              autoEscape={false}
              caseSensitive={false}
              textToHighlight={edge.node.description}
            />

          </p>

        </div>

        { !(this.props.visibility === 'local') &&
          <div
            data-tooltip={`${this.props.visibility}`}
            className={`Tooltip LocalLabbookPanel__${this.props.visibility}`}>
          </div>
        }
    </Link>)
  }
}
