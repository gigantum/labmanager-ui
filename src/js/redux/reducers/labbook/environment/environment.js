/**
 * constants
 */
export const CLOSE_ENVIRONMENT_MENUS = 'CLOSE_ENVIRONMENT_MENUS'
export const TOGGLE_PACKAGE_MENU = 'TOGGLE_PACKAGE_MENU'
export const TOGGLE_CUSTOM_MENU = 'TOGGLE_CUSTOM_MENU'
export const RESET_DETAIL_STORE = 'RESET_DETAIL_STORE'
export const CONTAINER_MENU_WARNING = 'CONTAINER_MENU_WARNING'
export const UPDATE_CONTAINER_MENU_VISIBILITY = 'UPDATE_CONTAINER_MENU_VISIBILITY'



export default (
 state = {
   'status': "",
   'containerMenuOpen': false,
   'containerMenuWarning': ''
 },
 action
) => {

if (action.type === CLOSE_ENVIRONMENT_MENUS) {
   return {
     ...state,
     packageMenuVisible: false,
     viewContainerVisible: false
   };
 }else if (action.type === TOGGLE_PACKAGE_MENU) {

   return {
     ...state,
     packageMenuVisible: action.payload.packageMenuVisible
   };
 }else if (action.type === TOGGLE_CUSTOM_MENU) {
   return {
     ...state,
     viewContainerVisible: action.payload.viewContainerVisible
   };
 }else if(action.type === RESET_DETAIL_STORE){
   return {
     ...state,
     detailMode: false
   };
 } else if(action.type === CONTAINER_MENU_WARNING) {
  return {
    ...state,
    containerMenuWarning: action.payload.message
  }
} else if (action.type === UPDATE_CONTAINER_MENU_VISIBILITY) {
  
  return {
    ...state,
    containerMenuOpen: action.payload.containerMenuOpen
  };
}

 return state;
};
