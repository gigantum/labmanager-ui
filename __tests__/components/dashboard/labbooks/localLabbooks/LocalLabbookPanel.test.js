import React from 'react'
import renderer from 'react-test-renderer';
import {shallow, mount} from 'enzyme';
import history from 'JS/history'
import {StaticRouter, Link} from 'react-router';
import json from './__relaydata__/LocalLabbooks.json'
import LocalLabbookPanel from 'Components/dashboard/labbooks/localLabbooks/LocalLabbookPanel';
import relayTestingUtils from 'relay-testing-utils'
import {MemoryRouter } from 'react-router-dom'
import environment from 'JS/createRelayEnvironment'
import {BrowserRouter as Router} from 'react-router-dom'

const variables = {first:5}
const goToLabbook = () =>{

}

const fixtures = {
  edge: json.data.labbookList.localLabbooks.edges[0],
  node: json.data.labbookList.localLabbooks.edges[0],
  key:'key',
  className: "LocalLabbooks__panel",
  history: history,
  environment: false,
  goToLabbook: goToLabbook
}

test('Test LocalLabbooks rendering', () => {

  const localLabbooks = renderer.create(

     relayTestingUtils.relayWrap(
      <Router>
        <LocalLabbookPanel
          {...fixtures}
        />
      </Router>, {}, json.data.labbookList.localLabbooks.edges[0])

  );

  const tree = localLabbooks.toJSON()

  expect(tree).toMatchSnapshot()
})

export default variables
