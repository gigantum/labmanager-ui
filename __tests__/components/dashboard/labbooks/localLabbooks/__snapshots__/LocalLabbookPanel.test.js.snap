// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test LocalLabbooks rendering 1`] = `
<a
  className="Card Card--text column-4-span-3 flex flex--column justify--space-between"
  href="/projects/cbutler/zab"
  onClick={[Function]}
>
  <div
    className="LocalLabbooks__row--icons"
  >
    <div
      className="LocalLabbooks__containerStatus"
    >
      <button
        className="ContainerStatus__container-state LocalLabbooks__containerStatus--state loading box-shadow"
        onClick={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
      >
        
      </button>
    </div>
  </div>
  <div
    className="LocalLabbooks__row--text"
  >
    <div>
      <h6
        className="LocalLabbooks__panel-title"
        onClick={[Function]}
      >
        <span
          className={undefined}
        >
          <span
            className=""
            style={undefined}
          >
            zab
          </span>
        </span>
      </h6>
    </div>
    <p
      className="LocalLabbooks__paragraph LocalLabbooks__paragraph--owner "
    >
      Created by cbutler
    </p>
    <p
      className="LocalLabbooks__paragraph LocalLabbooks__paragraph--description"
    >
      <span
        className={undefined}
      />
    </p>
  </div>
  <div
    className="Tooltip LocalLabbookPanel__undefined"
    data-tooltip="undefined"
  />
</a>
`;
